### DISCUSSION TITLE
share adaptive mesh from master app to sub app

### DISCUSSION THREAD

**Discussion by BoZeng1997** [idaholab/moose#16389] [Q&A General] [answered]
I am trying to clone the adaptive mesh from master app to sub app. I have set the **_clone_master_app=true_**  in block [Multiapps] for the sub app in the **master.i** file.  What should I do for the [Mesh] block in **sub.i**? I tried leaving it empty or add the original mesh file as  FileMesh to read. Either way returned with error: Use either FileMesh with an Exodus mesh file or FileMeshGenerator with an Exodus mesh file and with use_for_exodus_restart equal to true

**Comment by hugary1995** [comment, marked as answer]
Hi Bo, see the discussion [here](https://groups.google.com/d/msgid/moose-users/CACL%2BgcMfv-TXRxDJvBwz0zy-U6M5SHot%2BTwXcWkE5k%2B8mxZ_Vg%40mail.gmail.com?utm_medium=email&utm_source=footer) .

**Comment by aeslaughter** [comment]
We have some basic capability in the level set module for doing this. Check out the tests here: https://github.com/idaholab/moose/tree/next/modules/level_set/test/tests/transfers/markers
